### PR TITLE
Bump astroid to 3.3.1

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
Bump astroid to 3.3.1

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -40,7 +40,7 @@ dependencies    = [
     # Also upgrade requirements_test_min.txt.
     # Pinned to dev of second minor update to allow editable installs and fix primer issues,
     # see https://github.com/pylint-dev/astroid/issues/1341
-    "astroid>=3.2.4,<=3.3.0-dev0",
+    "astroid>=3.3.0,<=3.3.0-dev0",
```
Comment (DanielNoord) [inline pyproject.toml:43]:
```suggestion
    "astroid>=3.3.0,<=3.4.0-dev0",
```

**Review by DanielNoord — APPROVED** [review]

File: tests/primer/packages_to_prime.json [hunk 2]
```diff
@@ -28,6 +28,9 @@
   "music21": {
     "branch": "master",
     "directories": ["music21"],
+    "pylint_additional_args": [
+      "--disable=invalid-name,bad-docstring-quotes,docstring-first-line-empty"
```
Comment (jacobtylerwalls) [inline tests/primer/packages_to_prime.json:32]: I work on this package, so I know it doesn't use pylint for invalid-name. The additional f-string inferences results must just be swamping the output. It uses single quotes for docstrings, too. We don't need thousands of duplicative messages slowing things down.

**Comment by jacobtylerwalls** [comment]
Blocked by https://github.com/pylint-dev/astroid/issues/2492

File: tests/primer/packages_to_prime.json (hunk 2 above)
Comment (Pierre-Sassoulas) [reply tests/primer/packages_to_prime.json:32]: Isn't a project using single quote the only way to test thé Bad docstrings' quote message ?

Comment (jacobtylerwalls) [reply tests/primer/packages_to_prime.json:32]: I wrote that comment (and pushed this change) before verifying it was the cause of the failed run. Perhaps it's not.

I will likely remove it.

But to your point, something so simple can be tested with a unit test.

Comment (Pierre-Sassoulas) [reply tests/primer/packages_to_prime.json]: Sure, maybe this check is simple enough for a unit test (didn't check the code but it's likely). But the point of the primers is to encounter and test on a lot of unexpected code constructs to detect crashes without having to think about a relevant unit test first. If this check is simple/generic enough, why not disable it also for all primer repositories ? (For 'double quote docstrings repo' where it's never going to raise in particular).

**Review by DanielNoord — APPROVED** [review]

**Pull request merged**